mpapt-runtime/src/main/java/de/jensklingenberg/mpapt/extension/DeclarationCheckerImpl.kt

```diff
@@ -217,20 +218,13 @@ class DeclarationCheckerImpl(val processor: AbstractProcessor) : DeclarationChec
 
     private fun checkAnnotatedExpression(declaration: KtDeclaration, context: DeclarationCheckerContext, annotationName: String, roundEnvironment: RoundEnvironment, descriptor: SimpleFunctionDescriptor) {
 
-        /**
-         * We need to distinguish here between native and other platforms because the native compiler
-         * was always crashing when "...bodyExpression?.children?" was called.
-         * TODO: Check if versions >1.3.41 are still crashing
-         */
-
-        val annotatedExpressions = when (processor.activeTargetPlatform.first().platformName) {
-            KotlinPlatformValues.NATIVE -> {
-                ((declaration as KtNamedFunction).bodyExpression as KtBlockExpression).statements.filterIsInstance<KtAnnotatedExpression>()
-            }
-            else -> {
-                (declaration as KtNamedFunction).bodyExpression?.children?.filterIsInstance<KtAnnotatedExpression>()
-            }
-        }
+        val annotatedExpressions =
+                (declaration as KtNamedFunction)
+                        .bodyExpression
+                        ?.blockExpressionsOrSingle()
+                        ?.filterIsInstance<KtAnnotatedExpression>()
+                        ?.filterIsInstance<KtAnnotatedExpression>()
```

Comment:
twice?

Reply:
Whoops. No that shouldn't be there.
Probably misspressed Ctrl+S with Ctrl+D. Sorry about that.
I'll fix it once im home.